packages/core/src/v3/utils/flattenAttributes.ts

```
@@ -95,36 +95,47 @@ export function unflattenAttributes(
-        acc.push(...subparts);
-      } else {
-        acc.push(part);
-      }
-      return acc;
-    }, [] as string[]);
+    const parts = key.split(".").reduce(
+      (acc, part) => {
+        if (part.startsWith("[") && part.endsWith("]")) {
+          // Handle array indices more precisely
+          const match = part.match(/^\[(\d+)\]$/);
+          if (match && match[1]) {
```

Biome

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)